src/cache.rs

```diff
@@ -61,17 +63,29 @@ impl Cache {
     pub fn get_cache_dir() -> Result<(PathBuf, PathSource), TealdeerError> {
         // Allow overriding the cache directory by setting the
         // $TEALDEER_CACHE_DIR env variable.
-        if let Ok(value) = env::var("TEALDEER_CACHE_DIR") {
+        if let Ok(value) = env::var(CACHE_DIR_ENV_VAR) {
             let path = PathBuf::from(value);
-
-            if path.exists() && path.is_dir() {
-                return Ok((path, PathSource::EnvVar));
+            let path_exists = path.exists();
+            if path_exists && !path.is_dir() {
```

Comment:
This could still be reduced to one syscall (see the thread from the last review):
```suggestion
            let metadata = path.metadata();
            let path_exists = metadata.is_ok();
            let path_is_dir = path_exists && metadata.unwrap().is_dir();
            if path_exists && !path_is_dir {
```

Reply:
I did not understand the last comment correctly. Sorry for that! Now it should look like you proposed.

However, the `unwrap()` in line 70 is only justified because of applied short-circuit evaluation. What do you think about being a bit more explicit by writing `let path_is_dir = metadata.map_or(false, |md| md.is_dir());`?

Reply:
No worries! I agree, the `unwrap` stands out a bit, I didn't really like the idea of checking whether the metadata is `Some` explicitly twice. Your proposed solution is probably still better than the `unwrap`. I played around with that though a bit, what do you think of

```rust
let (exists, is_dir) = path.metadata().map_or((false, false), |md| (true, md.is_dir()));
```

Pick whichever of the two solutions you like best. @dbrgn and I will look over it once more and I believe we can then merge this :tada:

Reply:
I like the idea with the tuple. 👍